action/protocol/staking/handlers.go

```diff
@@ -93,19 +87,41 @@ func (p *Protocol) handleCandidateRegister(ctx context.Context, act *action.Cand
 func (p *Protocol) handleCandidateUpdate(ctx context.Context, act *action.CandidateUpdate, sm protocol.StateManager) (*action.Receipt, error) {
 	actCtx := protocol.MustGetActionCtx(ctx)
 	blkCtx := protocol.MustGetBlockCtx(ctx)
-	c, err := getCandidate(sm, actCtx.Caller)
-	if err != nil {
-		return nil, err
+
+	// only owner can update candidate
+	if !p.inMemCandidates.ContainsOwner(actCtx.Caller) {
+		return nil, ErrInvalidOwner
+	}
+	c := p.inMemCandidates.GetByOwner(actCtx.Caller)
```

Comment:
can we can check if c is nil to check if owner exist?

Reply:
ok